mono/metadata/assembly.c

```diff
@@ -558,6 +558,23 @@ mono_assembly_getrootdir (void)
 }
 
 /**
+ * mono_native_getrootdir:
+ * 
+ * Obtains the root directory used for looking up native libs (.so, .dylib).
+ *
+ * Returns: a string with the directory, this string should not be freed.
+ */
+G_CONST_RETURN gchar *
+mono_native_getrootdir (void)
+{
+	gchar* fullpath = g_malloc(strlen(mono_assembly_getrootdir ()) + strlen(mono_config_get_reloc_lib_dir()) + 1);
+	strcpy( fullpath, mono_assembly_getrootdir () );
+	strcat( fullpath, "/" ) ;
+	strcat( fullpath, mono_config_get_reloc_lib_dir() ) ;
+	return fullpath;
```

Comment:
I tried to find a path builder helper, and failed. Seems I was in the wrong bit of the GLib manual.